integration/tests/cook/test_basic.py

```diff
@@ -277,22 +289,20 @@ def test_multiple_progress_updates_submit(self):
         message = json.dumps(job['instances'][0], sort_keys=True)
         self.assertEqual('success', job['instances'][0]['status'], message)
 
-        job = util.wait_for_sandbox_directory(self.cook_url, job_uuid)
-        message = json.dumps(job['instances'][0], sort_keys=True)
-        self.assertIsNotNone(job['instances'][0]['output_url'], message)
-        self.assertIsNotNone(job['instances'][0]['sandbox_directory'], message)
+        instance = util.wait_for_sandbox_directory(self.cook_url, job_uuid)
+        message = json.dumps(instance, sort_keys=True)
+        self.assertIsNotNone(instance['output_url'], message)
+        self.assertIsNotNone(instance['sandbox_directory'], message)
 
-        if job_executor_type == 'cook':
+        if instance['executor'] == 'cook':
             util.sleep_for_publish_interval(self.cook_url)
-
-            job = util.wait_for_exit_code(self.cook_url, job_uuid)
-            message = json.dumps(job['instances'][0], sort_keys=True)
-            self.assertEqual(0, job['instances'][0]['exit_code'], message)
-
-            job = util.load_job(self.cook_url, job_uuid)
-            message = json.dumps(job['instances'][0], sort_keys=True)
-            self.assertEqual(75, job['instances'][0]['progress'], message)
-            self.assertEqual('Seventy-five percent', job['instances'][0]['progress_message'], message)
+            instance = util.wait_for_exit_code(self.cook_url, job_uuid)
+            message = json.dumps(instance, sort_keys=True)
+            self.assertEqual(0, instance['exit_code'], message)
+            self.assertEqual(75, instance['progress'], message)
+            self.assertEqual('Seventy-five percent', instance['progress_message'], message)
+        else:
+            self.logger.info(f'Bailing out because the cook executor is not being used: {instance}')
```

Comment:
Should we say `exit code and progress info not checked as cook executor was not used for {instance}`?

Reply:
Done